plugins/meta/portmap/portmap_nftables.go

```diff
@@ -252,14 +270,8 @@ func (pmNFT *portMapperNFTables) checkPorts(config *PortMapConf, containerNet ne
 			if isV6 != isHostV6 {
 				continue
 			}
-			if hostIP.IsUnspecified() {
-				hostPorts++
-			} else {
-				hostIPHostPorts++
-			}
-		} else {
-			hostPorts++
```

Comment:
OK, suppose you have a pod with hostIP hostports. They get created in `hostip_hostports` by the old plugin. Then you upgrade your plugins. Then the CRI does a CHECK. With this code, the CHECK would return an error. So then, presumably, the CRI will ADD again, and then `forwardPorts` will duplicate the hostIP rules in the `hostports` chain, without deleting the ones from the `hostip_hostports` chain...

That's not awful though; the extra rules are redundant but not incorrect. And it would be a little bit annoying to make `checkPorts` try to deal with accepting either the old layout or the new.

(Or would CRI be expected to do a DEL before the ADD anyway?)

Reply:
> (Or would CRI be expected to do a DEL before the ADD anyway?)

Yes, you must always do a DEL before ADD.